src/JsonSchema/Constraints/ConstraintInterface.php

```diff
@@ -59,5 +59,10 @@ public function isValid();
      * @param mixed            $i
      * @throws \JsonSchema\Exception\ExceptionInterface
      */
+    public function _check(&$value, $schema = null, JsonPointer $path = null, $i = null);
```

Comment:
Well, right, but this isn't a situation where you have to rely on the underscore convention to suggest visibility (like we often do in javascript), because we have actual visibility keywords. I think most people would agree it's sort of poor form to make a method public, but suggest that client code shouldn't call it with the underscore convention. Either it's private and you can't call it, or it's public and you can. Take a stand!